In [11]:
# Memuat Dataset Titanic
import pandas as pd
import numpy as np

URL = "https://raw.github.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)

df

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
...,...,...,...,...,...,...,...,...,...,...,...,...
886,887,0,2,"Montvila, Rev. Juozas",male,27.0,0,0,211536,13.0000,NaN,S
887,888,1,1,"Graham, Miss. Margaret Edith",female,19.0,0,0,112053,30.0000,B42,S
888,889,0,3,"Johnston, Miss. Catherine Helen ""Carrie""",female,NaN,1,2,W./C. 6607,23.4500,NaN,S
889,890,1,1,"Behr, Mr. Karl Howell",male,26.0,0,0,111369,30.0000,C148,C


In [14]:
# Q1 : Jumlah baris dan kolom dataset Titanic
print(df.shape, df.columns.tolist())


(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


Dataset Titanic memiliki 891 baris dan 12 kolom yang terdiri dari : PassengerId, Survived, Pclass, Name, Sex, Age, SibSp, Parch, Ticket, Fare, Cabin, Embarked

In [31]:
# Q2 : kolom mana yang memiliki nilai kosong (missing) pada dataset Titanic? berapa dan proprosinya
print(df.isnull().sum())
print(df.isnull().mean())


PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64
PassengerId    0.000000
Survived       0.000000
Pclass         0.000000
Name           0.000000
Sex            0.000000
Age            0.198653
SibSp          0.000000
Parch          0.000000
Ticket         0.000000
Fare           0.000000
Cabin          0.771044
Embarked       0.002245
dtype: float64


Dataset Titanic memiliki 3 kolom kosong (missing), Yaitu:

1. **Age** dengan jumlah 177 dan proporsi 0,19%
2. **Cabin** dengan jumlah 687 dan proporsi 0,77%
3. **Embarked** dengan jumlah 2 dan proporsi 0,02%





In [72]:
# Q3 : Berapa persen penumpang yang selamat (Survived=1)
selamat = df[df['Survived']==1]
persentase = len(selamat)/len(df)

print ("Total Penumpang :", len(df))
print("Penumpang selamat :", df['Survived'].sum())
print("Persentase selamat (%) :", persentase * 100)

Total Penumpang : 891
Penumpang selamat : 342
Persentase selamat (%) : 38.38383838383838


Persentase penumpang selamat sejumlah **38.38%** dari total penumpang

In [83]:
# Q4: Rata-rata usia penumpang wanita kelas 1 yang selamat
wk1_selamat = df[(df['Sex']=='female') &
                 (df['Pclass']==1) &
                 (df['Survived']==1)]
print ("Rata-Rata Usia Wanita Kelas I selamat :", wk1_selamat['Age'].mean())

Rata-Rata Usia Wanita Kelas I selamat : 34.9390243902439


Berdasarkan Dataset Titanic Rata-rata Usia Penumpang Kelas I yang selamat adalah 35 Tahun

In [88]:
# Q5: Kelas penumpang yang memiliki tingkat keselamatan tertinggi
hasil = df.groupby('Pclass').agg(
        total=('PassengerId', 'count'),
        selamat=('Survived','sum'),
        persen = ('Survived','mean'),
).round(3)
print(hasil)


        total  selamat  persen
Pclass                        
1         216      136   0.630
2         184       87   0.473
3         491      119   0.242


Tingkat keselamatan penumpang tertinggi terdapat pada kelas I yaitu **sejumlah** **136 penumpang** atau 0,630%

In [89]:
# Q6: Tampilan hasil groupby('Pclass')['Survived'].mean()
df.groupby('Pclass')['Survived'].mean()

,Survived
Pclass,
1,0.629630
2,0.472826
3,0.242363
